added module-drivers for programming RFoF transmitter and receiver (though we will be using PAMs)

Richardsaeed opened the pull request:
This module-drivers folder contains my edits to the source code from the original found here: [`module-drivers`](https://gitlab.com/dsa-2000/asp/rfof/module-drivers). Can attempt to build the edited version on an Rpi4, if it does not work, follow the guideline I wrote in the README.md on the main branch- building from the forked repo as outlined in README works.

Changes:
• added support for i2c 
• added rpi Ftx/Frx classes to python.rs which allows us to communicate with both over i2c on an Rpi4

Richardsaeed committed:
intial push of module-drivers from a working rpi4- can attempt to clone onto a new Rpi4- if it does not work follow the README.md

Richardsaeed committed:
Fix: converted module-drivers from submodule to regular folder

christianhbye commented on module-drivers/LICENSE-APACHE:
is there a reason to have two licenses? MIT is usually my go-to if there aren't strong reasons to do something else

Richardsaeed replied:
Unsure- would have to ask Kiran, I don't see any mention regarding reasoning for dual-licensing in their module-drivers GitLab: [module-drivers](https://gitlab.com/dsa-2000/asp/rfof/module-drivers)

christianhbye reviewed (commented):
Could you write little script similar to your jupyter notebook that can be run from the command line without a GUI? That simply sets the attenuation or other parameters

Richardsaeed commented:
@christianhbye, yes I can do that! I may already have a script on a different rpi 👍🏽.

christianhbye commented:
I just realized this is mostly Kiran's code. Can you add to the README that this is a fork of their repo and what the main changes you've made are?

Richardsaeed commented:
changes are covered in the main README.md here [RFoF_module](https://github.com/EIGSEP/RFoF_module/tree/main)
Can add to module-drivers README as well.

Richardsaeed committed:
updated Cargo.toml to add linux-embedded-hal
updated python.rs, adding FtxPi/FrxPi classes which allows for setting parameters over i2c

Richardsaeed committed:
updated README.md

Richardsaeed committed:
minor changes

Richardsaeed committed:
moved the private module-drivers repo Wei and I worked with to this branch

Richardsaeed committed:
Revert "moved the private module-drivers repo Wei and I worked with to this branch"

This reverts commit 16efb7c9427a64f0d935759bfa319bba4482fbce.

Richardsaeed committed:
copying over private module-driver repo used for testing to branch.

Richardsaeed committed:
Update README.md

Richardsaeed committed:
restore transmitter.py after revert

Richardsaeed committed:
added additional functionality.

Richardsaeed committed:
added functionality so every time we run the script the attenuation does not have to be set...

Richardsaeed committed:
Merge branch 'main' into rfof-modules

christianhbye commented on scripts/transmitter.py:
This looks good. Not sure you need the try/except loops here though, If the code fails, then you'll get errors and the program will quit anyway, so I am wondering if having try/except adds any information. You might as well just let the program fail.

Richardsaeed replied:
Changes made.

Richardsaeed committed:
removed try/except statements & added brief docstring.
